emily/handler/src/database/entries/deposit.rs

```diff
@@ -540,3 +552,79 @@ impl DepositUpdatePackage {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn deposit_update_should_be_unnecessary_when_event_is_present() {
+        let event = DepositEvent {
+            status: StatusEntry::Accepted,
+            message: "".to_string(),
+            stacks_block_height: 1,
+            stacks_block_hash: "".to_string(),
+        };
+
+        let deposit = DepositEntry {
+            key: Default::default(),
+            version: 0,
+            recipient: "".to_string(),
+            amount: 0,
+            parameters: Default::default(),
+            status: Status::Pending,
+            reclaim_script: "".to_string(),
+            deposit_script: "".to_string(),
+            last_update_height: 0,
+            last_update_block_hash: "".to_string(),
+            fulfillment: None,
+            history: vec![
+                DepositEvent {
```

Comment:
For readability I would probably create this in a variable above next to the accepted event and have this be `vec![pending, accepted]`. Took me a minute to realize that the `event.clone()` was there :see_no_evil: